src/Compiler/pars.fsy

```diff
@@ -665,6 +665,17 @@ moduleSpfn:
         let trivia: SynModuleSigDeclNestedModuleTrivia = { ModuleKeyword = Some mModule; EqualsRange = $4 }
         SynModuleSigDecl.NestedModule(info, isRec, $5, m, trivia) }
 
+  | opt_attributes opt_access moduleIntro error
```

Comment:
@kerams Could you explain why `recover` was changed to `error` in subsequent commits? They're mostly the same, expect for `recover` also recovers `EOF`: https://github.com/dotnet/fsharp/blob/d0dfdb9f1963cd42c28c2e1a673ded8e5659671c/src/Compiler/pars.fsy#L3447-L3456

Reply:
Because thanks to the `EOF` terminal, a file with just `module A` would get parsed as a nested module.

Reply:
Ah, indeed, thank you.

So we have two different scenarios:
* a new file with only one `module A` on top, and we don't want to parse it as a nested module
* an existing file, where you type a new module at the end and want to have a good recovery too (so lenses and highlightings don't blink/jump while you type it)

That could be an interesting thing to solve one day too! 🙂

Reply:
By the way, the LexFilter changes are still needed, otherwise we're pushing incorrectly pushing `CtxtModuleBody` on `module C` without `=`.